app/partials/desktop/confirm_modal.html

```diff
@@ -25,6 +25,15 @@
         <ng-pluralize count="count"
                      when="confirm_modal_X_messages_delete">
         </ng-pluralize>
+        <a class="tg_checkbox" ng-if="revocable" ng-click="data.revoke = !data.revoke" ng-class="data.revoke ? 'tg_checkbox_on' : ''">
```

Comment:
The checkbox should be replaced with two submit buttons. One is "Delete for me", second is "Delete for {peer_name}" / "Delete for all participants" (for groups).
Sometimes only one button is allowed, sometimes both. This setting should be passed to modal just like `revokable`, but may be with `deleteForMyself`, `deleteForAll` + `deleteForPeerID` parameters.